numpy/random/generator.pyx

```diff
@@ -3459,6 +3492,16 @@ cdef class Generator:
         final_shape.append(mean.shape[0])
         x = self.standard_normal(final_shape).reshape(-1, mean.shape[0])
 
+        if use_factor:
+            # check if the factor supplied is a cholesky upper triangular
+            # matrix and calculate random values accordingly.
+            if method == 'cholesky' and not np.any(np.tril(cov, k=-1)):
```

Comment:
Don't the other methods produce symmetric decompositions so that `np.dot(x, cov.T) == np.dot(x, cov)`, which would allow the paths to be simplified?

Reply:
No, that is not the case. The decompositions produced by SVD and Eigendecomposition need not be symmetric and thus `np.dot(x, cov.T)` is not necessarily equal to `np.dot(x, cov)`. Also in Cholesky decomposition if one uses the lower triangular form you get `L @ L.T = cov`, whereas if you use the upper triangular form you get `U.T @ U = cov`. Thus when supplying a factor matrix, it matter which method was used to calculate it.

The only way I think it can be simplified to a point where a method used doesnt matter is if the user is forced to use factor `F` in the form such that `F @ F.T = cov`

Reply:
Why not dictate to users that they must use the lower Cholesky factor (which is the default)?  Doesn't that allow this to be simplified and the keyword to be ignored completely when using factor?

Reply:
> Why not dictate to users that they must use the lower Cholesky factor (which is the default)? Doesn't that allow this to be simplified and the keyword to be ignored completely when using factor?

Correct, it would simplify things. I didn't want to enforce that assumption in case the user is ignorant of the input. Let me push a commit to address that. Thanks

Reply:
Also, the Cholesky  may not work in some situations so people may want to use the robust SVD method.

Providing the pre-computed SVD factors is an effective way to mitigate the slowness of the method - if you already have the SVD factors for other purposes or you need to make multiple calls to the method it's extremely wasteful to re-compute them.